### PR TITLE
feat(ci): add cfg check

### PR DESCRIPTION
**Pull request by emhane**
Checks if `cfg` attribute is used correctly. This for example catches if a feature is used in lib, that isn't in manifest.

### CODE REVIEW

**Review by Rjected — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -414,8 +414,11 @@ test:
 	make test-doc && \
 	make test-other-targets
 
+cfg-check:
+	cargo +nightly -Zcheck-cfg c
+
 pr:
-	make fmt && \
+	make cfg-check && \
```
Comment (onbjerg) [inline Makefile:421]: please add fmt back as well

File: .github/workflows/features.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  pull_request:
+  merge_group:
+  push:
+    branches: [main]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+    cfg-check:
```
Comment (onbjerg) [inline .github/workflows/features.yml:13]: lets put this in the ci workflow

**Review by onbjerg — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (emhane) [reply Makefile:421]: it's already in `make lint`

File: .github/workflows/features.yml (hunk 2 above)
Comment (emhane) [reply .github/workflows/features.yml:13]: wdym? suggestion pls

Comment (onbjerg) [reply .github/workflows/features.yml:13]: put it in https://github.com/paradigmxyz/reth/blob/main/.github/workflows/lint.yml and add it to `needs` for `lint-success`

Comment (emhane) [reply .github/workflows/features.yml:13]: thanks

**Comment by emhane** [comment]
thanks @DaniPopes for being a rust encyclopaedia and telling us ab this awesome check, will save me one or two fml moments here-forth when I forget to add `optimism` feature to a crate cause I'm so used to it being everywhere

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -126,11 +126,20 @@ jobs:
         with:
           cmd: jq empty etc/grafana/dashboards/overview.json
 
+  check-cfg:
+    name: check-cfg
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@nightly
```
Comment (onbjerg) [inline .github/workflows/lint.yml:135]:
```suggestion
      - uses: dtolnay/rust-toolchain@nightly
      - uses: Swatinem/rust-cache@v2
        with:
          cache-on-failure: true
```

i'd add caching

**Review by onbjerg — APPROVED** [review]
lgtm otherwise

**Review by mattsse — APPROVED** [review]

**Pull request merged**